### PR TITLE
Include users in groups and vice versa. List users: Allow filtering by groups.

### PR DESCRIPTION
**Pull request by nileshgulia1**
Right now we didn't have any way to convey membership between a user and group in volto. This will also allow groups to communicate on the basis of inherited roles of users they exhibit.

And fixes https://github.com/plone/plone.restapi/issues/1419

### CODE REVIEW

**Comment by nileshgulia1** [comment]
@jenkins-plone-org please run jobs

**Comment by nileshgulia1** [comment]
@jenkins-plone-org please run jobs

**Comment by ksuess** [comment]
@nileshgulia1 , I'll work on 'Add user to group' https://github.com/plone/volto/issues/3018 during Beethoven sprint. 
Can I work on this plone.restapi PR, or do you prefer that I branch it? Or do you continue working on this by your self? Please let me know, thank you. Of course we can team up for user control panel. When are you available during sprint days?

**Comment by nileshgulia1** [comment]
@ksuess No problem, you can take it. I was expecting "if" any GSoC student could handle it as a part of the whole "User management" project in this year's list of ideas. But I don't see it happening either. 

I will be happy to pair up, and will be available on 4th and 5th May during sprint days.

**Comment by nileshgulia1** [comment]
@jenkins-plone-org please run jobs

**Comment by nileshgulia1** [comment]
@jenkins-plone-org please run jobs

**Comment by ksuess** [comment]
Hi @nileshgulia1 you asked to rebase https://github.com/plone/plone.restapi/pull/1425. Are you ready with this one? Can I rebase? 
And, I am really sorry for messing this up. As I understood before the sprint, this is on hold and I can add my needed changes. Afterwards I decided to come up with a PR with just my stuff to have a smaller understandable PR, and not mixing to much in one. Now I am stuck, cause I need [this](https://github.com/plone/plone.restapi/pull/1425) for https://github.com/plone/volto/issues/3018 If you would merge yours, I can fix my rebase of https://github.com/plone/plone.restapi/pull/1425 Sorry again.

**Comment by ksuess** [comment]
Hi @tisto Do you think you can spare some time to review? Then the user group membership management https://github.com/plone/volto/issues/3018 can be finished afterwards.

**Comment by ksuess** [comment]
@jenkins-plone-org please run jobs

**Review by nileshgulia1 — COMMENTED** [review]
👍🏻

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/http-examples/users_filtered_by_groups.req [hunk 1]
```diff
@@ -0,0 +1,3 @@
+GET /plone/@users?groups_filter=Reviewers,Administrators HTTP/1.1
```
Comment (tisto) [inline src/plone/restapi/tests/http-examples/users_filtered_by_groups.req:1]: @nileshgulia1 @ksuess according to our naming convention we should use a dash in "group-filters".

https://plonerestapi.readthedocs.io/en/latest/conventions.html?highlight=parameters#naming-convention-for-attribute-names-in-uris

The way we pass lists/arrays in a URL parameter is also something that is not fully clear right now. I currently can't find the discussion, just this:

https://github.com/plone/plone.restapi/issues/45

@sneridagh @jackahl do you recall our discussion regarding passing arrays as URL param? That was in the GET vs POST discussion on the search endpoint, right?

Comment (ksuess) [reply src/plone/restapi/tests/http-examples/users_filtered_by_groups.req]: Renaming of attribute groups_filter to groups-filter done.
https://github.com/plone/plone.restapi/pull/1325/commits/bc42a349bb7740d7a61155d4f0c50c92a66c5c9f

**Comment by ksuess** [comment]
once again. seems to be fixed.

@jenkins-plone-org please run jobs

**Comment by ksuess** [comment]
@tisto @alin @sneridagh @erral @jensens @supersuess  **User Group Mangement**:  I am ftired to ask again: please review, the user group management is ready for reviewing: https://github.com/plone/volto/pull/3349

**Comment by nileshgulia1** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/http-examples/users_filtered_by_groups.req (hunk 1 above)
Comment (ksuess) [reply src/plone/restapi/tests/http-examples/users_filtered_by_groups.req]: groups-filter parameter now expected with colon list separator.
https://github.com/plone/plone.restapi/pull/1325/commits/3a32cb809af2384f0af155e23f6a19014b4040f4

**Comment by ksuess** [comment]
@jenkins-plone-org please run jobs

**Review by ksuess — APPROVED** [review]
awesome ;-)

**Comment by nileshgulia1** [comment]
@ksuess Thank you for your efforts! 🎉

**Pull request merged**